service/frontend/service.go

```diff
@@ -61,6 +61,13 @@ type Config struct {
 	PersistencePerShardNamespaceMaxQPS    dynamicconfig.IntPropertyFnWithNamespaceFilter
 	EnablePersistencePriorityRateLimiting dynamicconfig.BoolPropertyFn
 
+	EnableDynamicRateLimiting               dynamicconfig.BoolPropertyFn
+	DynamicRateLimitingRefreshInterval      dynamicconfig.DurationPropertyFn
+	DynamicRateLimitingLatencyThreshold     dynamicconfig.FloatPropertyFn
+	DynamicRateLimitingErrorThreshold       dynamicconfig.FloatPropertyFn
+	DynamicRateLimitingRateBackoffStepSize  dynamicconfig.FloatPropertyFn
+	DynamicRateLimitingRateIncreaseStepSize dynamicconfig.FloatPropertyFn
```

Comment:
If you have a common set of dynamic config properties that are all used together, consider: define a struct for them, use a dynamic config "map" (which gives you a `map[string]any`) and then write a function to unpack the map into the struct (slightly annoying but you can cheat and do json.Marshal+json.Unmarshal to get it for free)

Reply:
@dnr I updated the dynamic config properties, let me know what you think.

I'm a little concerned about two things:
1. Will `json.Marshal` and `json.Unmarshal` add significant overhead when updating the dynamic values?
2. The field names to use in that map are somewhat harder to find. I included them on the comment on the constant for `dynamicconfig.HistoryPersistenceDynamicRateLimitingParams`, do you think that will be clear enough or will users have trouble understanding what config options are available?